### DISCUSSION TITLE
Cannot find module '@radix-ui/react-compose-refs' on NextJS

### DISCUSSION THREAD

**Discussion by hugoalvarenga** [shadcn-ui/ui#947] [General]
Hello!

I have a problem with components that have some kind of interaction, such as select, table, etc. I am using a Next 13 application.

It keeps returning the following error:

![image](https://github.com/shadcn/ui/assets/19660517/9b154dd3-3186-4442-99ef-77f69b892de6)

I've already uninstalled all shadcn-ui and radix-ui components, I've deleted the node_modules folder, I've tried to update the components, but without success.

`    "@radix-ui/react-accessible-icon": "^1.0.3", 
    "@radix-ui/react-accordion": "^1.1.2",
    "@radix-ui/react-alert-dialog": "^1.0.4",
    "@radix-ui/react-aspect-ratio": "^1.0.3",
    "@radix-ui/react-avatar": "^1.0.3",
    "@radix-ui/react-checkbox": "^1.0.4",
    "@radix-ui/react-collapsible": "^1.0.3",
    "@radix-ui/react-compose-refs": "^1.0.1",
    "@radix-ui/react-context-menu": "^2.1.4",
    "@radix-ui/react-dialog": "^1.0.4",
    "@radix-ui/react-dropdown-menu": "^2.0.5",
    "@radix-ui/react-hover-card": "^1.0.6",
    "@radix-ui/react-icons": "^1.3.0",
    "@radix-ui/react-label": "^2.0.2",
    "@radix-ui/react-menubar": "^1.0.3",
    "@radix-ui/react-navigation-menu": "^1.1.3",
    "@radix-ui/react-popover": "^1.0.6",
    "@radix-ui/react-progress": "^1.0.3",
    "@radix-ui/react-radio-group": "^1.1.3",
    "@radix-ui/react-scroll-area": "^1.0.4",
    "@radix-ui/react-select": "^1.2.2",
    "@radix-ui/react-separator": "^1.0.3",
    "@radix-ui/react-slider": "^1.1.2",
    "@radix-ui/react-slot": "^1.0.2",
    "@radix-ui/react-switch": "^1.0.3",
    "@radix-ui/react-tabs": "^1.0.4",
    "@radix-ui/react-toast": "^1.1.4",
    "@radix-ui/react-toggle": "^1.0.3",
    "@radix-ui/react-toggle-group": "^1.0.4",
    "@radix-ui/react-tooltip": "^1.0.6",
    "@tanstack/react-table": "^8.9.1",
    "next": "^13.4.10",
`

How to solve this problem?

**Comment by corysimmons** [comment]
If that's your `package.json`, I'd suggest you also post your `yarn.lock` file. It might give more insight into what's actually missing.

`react-slot` looks like it requires `react-compose-refs`, but it can't find it for some reason. This might be a problem with Radix instead of ShadcnUI. Specifically, that version of `react-slot` (assuming that's the culprit) has probably been patched in the 3 weeks since you posted this, so try `yarn add @radix-ui/react-slot` again (I'm not sure if `react-slot` will auto upgrade `react-compose-refs`--I'd imagine it would since it's a dependent).
